src/config.hpp

```diff
@@ -170,9 +170,9 @@ class config
 		pointer operator->() const { return &**i_; }
 
 		bool operator==(const child_iterator &i) const { return i_ == i.i_; }
-		bool operator!=(const child_iterator &i) const { return i_ != i.i_; }
+		bool operator!=(const child_iterator &i) const { return !operator==(i); }
 		bool operator==(const const_child_iterator &i) const { return i == *this; }
-		bool operator!=(const const_child_iterator &i) const { return i == *this; }
+		bool operator!=(const const_child_iterator &i) const { return i != *this; }
```

Comment:
Prefer !operator== here, as you did just above.

The idea is changing == also changes !=